### ISSUE TITLE
Read Only features should be ignored

### ISSUE THREAD

**Issue by NoXPhasma** [Lurkki14/tuxclocker#2]
Tuxclocker crashes when I try to apply any changes. The cause seems to be the fact that Pascal cards don't support voltage changes on Linux:
```
ERROR: The attribute 'GPUOverVoltageOffset' specified in assignment 'GPUOverVoltageOffset=99999999' cannot be assigned
       (it is a read-only attribute).
ERROR: The attribute 'GPUOverVoltageOffset' specified in assignment 'GPUOverVoltageOffset=99999999' cannot be assigned
       (it is a read-only attribute).
[1]    32187 segmentation fault (core dumped)  ./tuxclocker
```

Tuxlocker should check if specific features are read only and ignore them on saving.

**Comment by Lurkki14** [comment]
I'm in the progress of rewriting nvidia controlling using libxnvctrl which can properly detect if attributes are writable. An update that fixes this should be out in less than a week.

**Comment by Lurkki14** [comment]
The dev branch now has detection for the readability of attributes, but only the sliders get disabled for now. You can try it and see if the sliders get greyed out for read-only features.

**Comment by NoXPhasma** [comment]
I tried, but I can't compile it. It's complaining about not no file `/opt/cuda/include/nvml.h`. No idea how I have to provide that, nvidia-cuda-dev is already installed.

**Comment by NoXPhasma** [comment]
I was able to build it, after editing the nvidia.h file and change the include path to `/usr/include/nvml.h`.

The voltage slider is disabled indeed.

**Comment by Lurkki14** [comment]
Thanks for the report, I'll make it use environment variables for the include path

**Comment by Lurkki14** [comment]
Should be fixed with today's release.

**Comment by NoXPhasma** [comment]
Yes, I can now save without a crash. However, the fan curve seems not to work. But for this it's probably better to open a new issue.
